core/state/iterator_test.go

```diff
@@ -85,9 +86,19 @@ func TestNodeIteratorCoverage(t *testing.T) {
 // database entry belongs to a trie node or not.
 func isTrieNode(scheme string, key, val []byte) (bool, common.Hash) {
 	if scheme == rawdb.HashScheme {
-		if len(key) == common.HashLength {
+		ok := rawdb.IsLegacyTrieNode(key, val)
```

Comment:
This check will, after we merge this (but not enable it), start checking that the hash of val corresponds to the key. What is the reason this check requires the extra strictness?

Reply:
Because this check is mostly used in `db inspect` command in this case performance is not that priority one thing.
I feel like the condition `key = 32byte` is too loose.

But I can change it back though.

Reply:
> Because this check is mostly used in db inspect command in this case performance is not that priority one thing.

Ah, yeah if that's indeed the case then it's fine by me.